### ISSUE TITLE
[Bug]:When insert/delete table with multiple blocks, and only one segment, errors occur.

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#584]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

1ca85c2e

### Actual behavior

When inserting or deleting a table with multiple blocks, but only one segment, some errors occur.
![image](https://github.com/infiniflow/infinity/assets/69421435/3ceddac2-ea76-4056-b1d6-61f4a534fa2a)

### Steps to reproduce

```Markdown
Function test code can be found in python/test/test_delete.py `test_delete_table_with_one_segment`
```

**Comment by Ognimalf** [comment]
Fixed by #614

**Comment by chrysanthemum-boy** [comment]
It looks fixed.
